### PR TITLE
Add org.gtaf.quran

### PR DESCRIPTION
**Pull request by SNNafi**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. 
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. 

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: pipewire/pipewire-jack-runtime-aarch64.pc
Comment (hfiguiere) [inline pipewire/pipewire-jack-runtime-aarch64.pc:1]: why do you need this?

File: pipewire/pipewire-jack-runtime.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "pipewire-jack-runtime",
```
Comment (hfiguiere) [inline pipewire/pipewire-jack-runtime.json:2]: this is in the runtime.

File: org.gtaf.quran.yml [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+      - ln -s /app/quran/qm_flutter /app/bin/qm_flutter
+      - mv data/gui/quran.desktop ${FLATPAK_ID}.desktop
+      - desktop-file-edit ${FLATPAK_ID}.desktop --set-icon=${FLATPAK_ID}
+      - install -Dm644 -t "${FLATPAK_DEST}/share/applications/" ${FLATPAK_ID}.desktop
+      - install -Dm644 data/gui/quran.svg "${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"
+      - install -Dm644 ${FLATPAK_ID}.appdata.xml /app/share/appdata/${FLATPAK_ID}.appdata.xml
+    sources:
+      - type: archive
+        dest: data
+        url: https://direct-download-apps.s3.us-east-2.amazonaws.com/quran/quran_0.5.7.tar.gz
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:278]: so how does this work on aarch64?

File: org.gtaf.quran.yml [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+          - https://mirrors.ocf.berkeley.edu/gnu/libcdio/libcdio-2.1.0.tar.bz2
+          - https://ftpmirror.gnu.org/gnu/libcdio/libcdio-2.1.0.tar.bz2
+        sha256: 8550e9589dbd594bfac93b81ecf129b1dc9d0d51e90f9696f1b2f9b2af32712b
+        x-checker-data:
+          type: html
+          url: https://ftp.gnu.org/gnu/libcdio/
+          version-pattern: libcdio-(\d\.\d+\.?\d*).tar.bz2
+          url-template: https://ftp.gnu.org/gnu/libcdio/libcdio-$version.tar.bz2
+
+  - name: libcdio-paranoia
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:94]: why do you need this?

File: org.gtaf.quran.yml [hunk 4]
```diff
@@ -0,0 +1,283 @@
+id: org.gtaf.quran
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:3]:
```suggestion
runtime-version: '23.08'
```

File: org.gtaf.quran.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            </image>
+        </screenshot>
+        <screenshot>
+            <caption>Translation, Word by Word, Tafsir &amp; more</caption>
+            <image>
+                https://store-images.s-microsoft.com/image/apps.36988.14497699918764450.bc9d91c0-96d6-4fd9-85e9-53359041c447.fccc9f1c-f0db-46da-9d68-2b9320d7f6f2
+            </image>
+        </screenshot>
+    </screenshots>
+    <content_rating type="oars-1.0" />
```
Comment (hfiguiere) [inline org.gtaf.quran.appdata.xml:211]:
```suggestion
    <content_rating type="oars-1.1" />
```

File: org.gtaf.quran.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+            <br />
+            "Whoever calls people to right guidance will have a reward like that of those who follow
+            him..." - Sahih Muslim, Hadith 2674
+
+            <br />
+        </p>
+        <p>
+            Developed by
+            <strong>Greentech Apps Foundation</strong>
+            <br />
```
Comment (hfiguiere) [inline org.gtaf.quran.appdata.xml:173]: this is not HTML. There is no `<br>`

File: org.gtaf.quran.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+            <br />
+            Website:
+            <a href="https://gtaf.org">https://gtaf.org</a>
```
Comment (hfiguiere) [inline org.gtaf.quran.appdata.xml:176]: href are NOT allowed in the description.

File: org.gtaf.quran.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+        <p>
+            ● Word by Word Audio Playback
+
+            <br />
+            <br />
+
+            <strong>Various Customization Options &amp; Others</strong>
+            <br />
+            <br />
+            ● Read in Uthmanic / Indopak script
```
Comment (hfiguiere) [inline org.gtaf.quran.appdata.xml:126]: use proper `<ul><li>`

File: pipewire/pipewire-jack-runtime-aarch64.pc
Comment (SNNafi) [reply pipewire/pipewire-jack-runtime-aarch64.pc:1]: Removed

File: org.gtaf.quran.yml (hunk 2 above)
Comment (SNNafi) [reply org.gtaf.quran.yml:278]: Added flathub.json

File: org.gtaf.quran.yml (hunk 3 above)
Comment (SNNafi) [reply org.gtaf.quran.yml:94]: To build the module `mpv`

File: org.gtaf.quran.yml (hunk 4 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Updated

File: org.gtaf.quran.appdata.xml (hunk 6 above)
Comment (SNNafi) [reply org.gtaf.quran.appdata.xml]: Fixed

File: org.gtaf.quran.appdata.xml (hunk 8 above)
Comment (SNNafi) [reply org.gtaf.quran.appdata.xml]: Fixed

File: org.gtaf.quran.appdata.xml (hunk 7 above)
Comment (SNNafi) [reply org.gtaf.quran.appdata.xml]: Fixed

**Comment by SNNafi** [comment]
@hfiguiere , Please check.

File: org.gtaf.quran.yml (hunk 3 above)
Comment (hfiguiere) [reply org.gtaf.quran.yml:94]: can't this be disabled?

Comment (SNNafi) [reply org.gtaf.quran.yml:94]: I have checked. Still can't figure out how to do this :|

File: org.gtaf.quran.yml [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+              type: git
+              tag-pattern: ^(\d{1,2}\.\d{1,2}\.\d{1,4})$
+
+  - name: mpv
+    buildsystem: meson
+    config-opts:
+      - -Dbuild-date=false
+      - -Dlibmpv=true
+      - -Dmanpage-build=disabled
+      - -Dcdda=enabled
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:218]: why do you need cdda?

Comment (SNNafi) [reply org.gtaf.quran.yml:218]: Oh. I have missed that part. I don't need this. Removing. Will commit soon. Let me build and check locally first. Thanks .

File: org.gtaf.quran.yml (hunk 3 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Removed

File: org.gtaf.quran.yml (hunk 9 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Removed

**Comment by SNNafi** [comment]
@hfiguiere,  please check. Thanks for your time.

File: org.gtaf.quran.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+        <p>
+            "Whoever calls people to right guidance will have a reward like that of those
+            who follow
+            him..." - Sahih Muslim, Hadith 2674
+
+        </p>
+        <p>Developed by
+            Greentech Apps Foundation
+        </p>
+        <p>Website: https://gtaf.org</p>
```
Comment (hfiguiere) [inline org.gtaf.quran.appdata.xml:106]: URLs are not allowed.

File: org.gtaf.quran.yml [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+    sources:
+      - type: git
+        url: https://github.com/libass/libass.git
+        tag: 0.17.1
+        commit: e8ad72accd3a84268275a9385beb701c9284e5b3
+        x-checker-data:
+          type: git
+          tag-pattern: ^(\d\.\d{1,3}\.\d{1,2})$
+
+  - name: libcdio
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:73]: this also should be removed.

File: org.gtaf.quran.appdata.xml (hunk 10 above)
Comment (SNNafi) [reply org.gtaf.quran.appdata.xml:106]: Anything else?

Comment (SNNafi) [reply org.gtaf.quran.appdata.xml:106]: Is everything alright other than these?

File: org.gtaf.quran.yml (hunk 11 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Removed

File: org.gtaf.quran.appdata.xml (hunk 10 above)
Comment (SNNafi) [reply org.gtaf.quran.appdata.xml]: Fixed

**Comment by SNNafi** [comment]
@hfiguiere , please give a check.

File: .DS_Store
Comment (bbhtt) [inline .DS_Store:1]: why is this here?

File: mpv [hunk 12]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+ln -snf "$(xdg-user-dir PICTURES)" "$HOME/.var/app/io.mpv.Mpv/Pictures"
+exec mpv-bin "$@"
```
Comment (bbhtt) [inline mpv:4]: what is this for?

File: org.gtaf.quran.yml [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+    build-commands:
+      - install -D mpv /app/bin/mpv
+    sources:
+      - type: file
+        path: mpv
+
+  #/ ### Python modules for mpv scripts
+
+  # This mpv plugin prevents screen blanking in GNOME (wayland) while playing media.
+  - name: mpv_inhibit_gnome
```
Comment (bbhtt) [inline org.gtaf.quran.yml:205]: not adding third party addons would be nice

File: org.gtaf.quran.yml [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+              type: git
+              tag-pattern: ^(\d{1,2}\.\d{1,2}\.\d{1,4})$
+
+  - name: mpv
+    buildsystem: meson
+    config-opts:
+      - -Dbuild-date=false
+      - -Dlibmpv=true
+      - -Dmanpage-build=disabled
+      - -Dsdl2=enabled
```
Comment (bbhtt) [inline org.gtaf.quran.yml:176]: Do you actually need the full mpv or just libmpv? Most of it looks copied from the MPV flatpak

Comment (SNNafi) [reply org.gtaf.quran.yml]: Removed

File: org.gtaf.quran.yml (hunk 13 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Removed

File: .DS_Store
Comment (SNNafi) [reply .DS_Store:1]: Added .gitignore

**Comment by bbhtt** [comment]
bot, build org.gtaf.quran

**Comment by bbhtt** [comment]
You need to build gslang too, the one is runtime is too old.

**Comment by SNNafi** [comment]
bot, build org.gtaf.quran

**Comment by SNNafi** [comment]
Got this error, `No GSettings schemas are installed on the system`. Built fine locally

**Comment by bbhtt** [comment]
> Got this error, `No GSettings schemas are installed on the system`. Built fine locally

Icon size is wrong

```
error: /srv/buildbot/worker/build-x86_64-5/build/builddir/export/share/icons/hicolor/scalable/apps/org.gtaf.quran.svg is not a valid icon: Expected a square icon but got: 401x400
```

it should be 401x401 or 400x400

**Comment by SNNafi** [comment]
bot, build org.gtaf.quran

**Comment by bbhtt** [comment]
Seems to work but I don't see mpv being used?

**Comment by SNNafi** [comment]
@bbhtt , We are using it only for playing audio for now.

**Comment by SNNafi** [comment]
So, What now? @bbhtt

**Comment by bbhtt** [comment]
It will be merged if everything is all right in a few days/weeks, please have patience.

**Comment by SNNafi** [comment]
Thanks

File: mpv [hunk 15]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+ln -snf "$(xdg-user-dir PICTURES)" "$HOME/.var/app/io.mpv.Mpv/Pictures"
```
Comment (hfiguiere) [inline mpv:3]: what is this supposed to achieve?

File: org.gtaf.quran.yml [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+    sources:
+      - type: git
+        url: https://github.com/mpv-player/mpv.git
+        tag: v0.37.0
+        commit: 818ce7c51a6b9179307950e919983e0909942098
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+
+  - name: mpv-wrapper
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:190]: this should be part of `mpv` module.

File: quran.svg
Comment (hfiguiere) [inline quran.svg:1]: There is a `data/gui/quran.svg` file in the tarball that is exactly this. So why is this one here?

File: org.gtaf.quran.yml [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+  - name: qm_flutter
+    buildsystem: simple
+    build-commands:
+      - cp -r data/quran /app/
+      - chmod +x /app/quran/qm_flutter
+      - mkdir -p /app/bin
+      - ln -s /app/quran/qm_flutter /app/bin/qm_flutter
+      - mv data/gui/quran.desktop ${FLATPAK_ID}.desktop
+      - desktop-file-edit ${FLATPAK_ID}.desktop --set-icon=${FLATPAK_ID}
+      - install -Dm644 -t "${FLATPAK_DEST}/share/applications/" ${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline org.gtaf.quran.yml:207]:
```suggestion
      - desktop-file-edit --set-icon=${FLATPAK_ID} data/gui/quran.desktop
      - install -Dm644 data/gui/quran.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```

no need to make it complicated.

File: mpv (hunk 15 above)
Comment (bbhtt) [reply mpv:3]: This should be removed. From what I saw, there was no way to screenshot using mpv in the app. The only part is a small popup audio player.

**Comment by SNNafi** [comment]
bot, build org.gtaf.quran

File: mpv (hunk 15 above)
Comment (SNNafi) [reply mpv]: Removed

File: quran.svg
Comment (SNNafi) [reply quran.svg:1]: Removed

File: org.gtaf.quran.yml (hunk 16 above)
Comment (SNNafi) [reply org.gtaf.quran.yml]: Moved under `mpv` module

**Comment by SNNafi** [comment]
Sorry for mentioning @hfiguiere. I don't know where to.

Currently, our app only supports __x86_64__ architecture. But I want to add support for __arm64__ also. But in that case, I need to build another executable that supports _arm64_. Then I will get two different executables for two different architectures.

But I don't understand and also didn't find any documentation, on how to add them in a single _.yml_ file and conditionally download them based on the architecture.

Can you please give me some idea in this regard or point me to some documentation regarding this? Thanks. Sorry again for mentioning.
